packages/engine/src/avatar/functions/createAvatar.ts

```diff
@@ -55,14 +125,21 @@ export const createAvatar = (spawnAction: typeof NetworkWorldAction.spawnAvatar.
         timer: 0
       })
   }
-  const transform = addComponent(entity, TransformComponent, {
-    position: new Vector3().copy(spawnAction.parameters.position),
-    rotation: new Quaternion().copy(spawnAction.parameters.rotation),
-    scale: new Vector3(1, 1, 1)
-  })
+
+  const position = createVector3Proxy(TransformComponent.position, entity).copy(spawnAction.parameters.position)
+
+  const rotation = createQuaternionProxy(TransformComponent.rotation, entity).copy(spawnAction.parameters.rotation)
+
+  // todo: figure out why scale makes avatar disappear
+  // const scale = createVector3Proxy(TransformComponent.scale, entity)
```

Comment:
Scale needs to be set to (1, 1, 1) by default, is the avatar disappearing because this line is setting all scale values to 0?

Reply:
strangely enough, it happens regardless of value! i just forgot to include `.copy(new Vector3(1, 1, 1))` in the commented out code

Reply:
this is something that we can figure out in the `threecs-next` repo